packages/engine/core/src/lib/abstract.ts

```diff
@@ -56,10 +56,10 @@ export class Plugin<T extends Api = any, App extends ApiMap = any> implements Pl
     this.profile.methods = methods
   }
 
-  activate() {
+  activate(): any | Promise<any> {
     if (this.onActivation) this.onActivation()
```

Comment:
shouldn't this `return` something?

Reply:
no it's a hook. It's not suppose to block the process.